src/main/scala/io/iohk/ethereum/blockchain/data/GenesisDataLoader.scala

```diff
@@ -0,0 +1,142 @@
+package io.iohk.ethereum.blockchain.data
+
+import akka.util.ByteString
+import io.iohk.ethereum.network.p2p.messages.PV63.AccountImplicits
+import io.iohk.ethereum.rlp.RLPList
+import io.iohk.ethereum.utils.{Config, Logger}
+import io.iohk.ethereum.{rlp, crypto}
+import io.iohk.ethereum.db.dataSource.{EphemDataSource, DataSource}
+import io.iohk.ethereum.db.storage.{Namespaces, NodeStorage}
+import io.iohk.ethereum.domain.{Block, BlockHeader, Account, Blockchain}
+import io.iohk.ethereum.mpt.{RLPByteArraySerializable, MerklePatriciaTrie}
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import io.iohk.ethereum.rlp.RLPImplicits._
+import org.spongycastle.util.encoders.Hex
+import spray.json._
+
+import scala.io.Source
+import scala.util.{Failure, Success, Try}
+
+class GenesisDataLoader(dataSource: DataSource, blockchain: Blockchain) extends Logger{
+
+  import GenesisDataLoader._
+  import JsonProtocol._
+  import AccountImplicits._
+
+  private val bloomLength = 512
+  private val hashLength = 64
+
+  private implicit val accountSerializer = new RLPByteArraySerializable[Account]
+
+  private val emptyTrieRootHash = ByteString(crypto.kec256(rlp.encode(Array.emptyByteArray)))
+  private val emptyEvmHash: ByteString = ByteString(crypto.kec256(Array.empty))
+
+  def loadGenesisData(): Unit = {
+    import Config.Blockchain.customGenesisFileOpt
+
+    log.info("Loading genesis data")
+
+    val genesisJson = customGenesisFileOpt match {
+      case Some(customGenesisFile) =>
+        log.debug(s"Trying to load custom genesis data from file: $customGenesisFile")
+        Try(Source.fromFile(customGenesisFile).getLines().mkString) match {
+          case Success(customGenesis) =>
+            log.info(s"Using custom genesis data from file: $customGenesisFile")
+            customGenesis
+          case Failure(ex) =>
+            log.error(s"Cannot load custom genesis data from file: $customGenesisFile", ex)
+            sys.exit(1)
```

Comment:
I don't like `sys.exit`. That's like `throw` or `return` in the functional paradigm. Non-blocking remark though.